src/Aws/General.hs

```diff
@@ -182,9 +183,10 @@ data Region
     | UsEast1
     | UsWest1
     | UsWest2
-    deriving (Show, Read, Eq, Ord, Enum, Bounded, Typeable)
+    | CustomEndpoint !T.Text
```

Comment:
Good point; in `aws-kinesis`, for instance, the port is hardcoded in (https://github.com/alephcloud/hs-aws-kinesis/blob/master/src/Aws/Kinesis/Core.hs#L241), but we would need to extract it from here. So I'll add another field.